ms/MSOper/MSMetaData.cc

```diff
@@ -3399,6 +3399,8 @@ void MSMetaData::_computeScanAndSubScanProperties(
     uInt chunkSize = nrows/nchunks;
     if (nrows % nchunks > 0) {
         ++chunkSize;
+        // integer division
+        nchunks = nrows/chunkSize + 1;
     }
```

Comment:
you would have less special cases to if you remove this `if` completely and change the lower loop to

    for (i=0; i < nrows; i+= chunkSize)
    ...
      i , std::min(i +chunkSize, nrows)